linera-core/src/unit_tests/worker_tests.rs

```diff
@@ -719,6 +722,7 @@ where
                 }
                 .into_hash()
                 .await,
+                oracle_records: vec![OracleRecord::default(); 2],
```

Comment:
Is it important to keep the record separate for each transaction?

Reply:
(see other comments)

Reply:
It did simplify the implementation, because you knew what portion to pass into each call.

I guess we could pass in a mutable reference to the iterator over all the block's oracle responses instead.

If we do, we should consider doing it in a different PR, because everything that touches the basic data types is complicated (affects GraphQL, the indexer, the RPC format, etc.).